File: python/diJet.py
class diJet:
    def __init__(self, p1, p2):
        self.p1 = p1
        self.p2 = p2

        self.p   = p1.p + p2.p
        self.dR  = p1.p.DeltaR(p2.p)
        self.m   = self.p.M()
        self.pt  = self.p.Pt()
        self.pt  = self.p.Pt()
        self.eta = self.p.Eta()
        self.phi = self.p.Phi()
        self.st  = self.p1.pt + self.p2.pt

        self.lead = self.p1 if self.p1.pt > self.p2.pt else self.p2
        self.subl = self.p2 if self.p1.pt > self.p2.pt else self.p1

    def dump(self):
        print( "diJet | mass",str(self.m).ljust(12),"| dR",str(self.dR).ljust(20),"| pt",str(self.pt).ljust(20),"| eta",str(self.eta).ljust(20),"| phi",str(self.phi).ljust(20) )
        print( "  lead "+self.lead.getDump() )
        print( "  subl "+self.subl.getDump() )

File: NtupleAna/src/dijet.cc
#include "ZZ4b/NtupleAna/interface/dijet.h"

using namespace NtupleAna;

//dijet object
dijet::dijet(){}
dijet::dijet(jet& jet1, jet& jet2){

  lead = (jet1.pt > jet2.pt) ? jet1 : jet2;
  subl = (jet1.pt > jet2.pt) ? jet2 : jet1;

  p   = jet1.p + jet2.p;
  dR  = jet1.p.DeltaR(jet2.p);
  st  = jet1.pt + jet2.pt;
  pt  = p.Pt();
  eta = p.Eta();
  phi = p.Phi();
  m   = p.M();
  e   = p.E();

  pZ  = p*(mZ/m);
  pH  = p*(mH/m);

}

dijet::~dijet(){}


